src/acl/external/file_userip/ext_file_userip_acl.cc

```diff
@@ -134,6 +144,10 @@ load_dict(FILE * FH) {
 
     }
 
+    if (current_entry) {
+        memset(current_entry, 0, sizeof *current_entry);
```

Comment:
Sometimes, we do not initialize some of the ip_user_dict entry fields inside the loop. Thus, this memsert() should be done immediately after each of the two malloc() calls or, if possible, both xmalloc() calls should be replaced with xcalloc() instead.

Or one can use C++ new/delete (with proper initialization), but that may require even more work.

The whole idea of allocating an "empty" trailing entry is flawed, but I do not know whether fixing that would require a lot of changes.